### PR TITLE
Where should my code live?

### PR DESCRIPTION
**Pull request by SiAdcock**
## What does this change?

Adds some high level contributor guidelines

## Why?

Provides contributors with some guidance on where their code should live

### CODE REVIEW

File: docs/contributing/where-should-my-code-live.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+They can be added to the `lowPriorityScripts` array in the [`packages/frontend/web/document.tsx`](https://github.com/guardian/dotcom-rendering/blob/master/packages/frontend/web/document.tsx) module.
+
+### High priority scripts
+
+High priority scripts are injected by JavaScript running in the `<head>` of the document. They are loaded asynchronously and executed in the order they are requested. 
+
+Examples include the [polyfill.io](https://polyfill.io) response and the main application JavaScript bundles.
+
+⚠️ **High priority scripts should be added sparingly.**
```
Comment (nicl) [inline docs/contributing/where-should-my-code-live.md]: I think it's probably worth saying at least two Dotcom team members need to approve any high priority scripts. (Or something similar, which effectively gatekeeps this to the Dotcom team.)

Comment (SiAdcock) [reply docs/contributing/where-should-my-code-live.md]: Sounds good, I'll make it clearer that adding another high priority script is unlikely to be approved!

**Comment by GHaberis** [comment]
@SiAdcock lovely instructions here. It may be we do this in a separate PR but what do you think about clarifying where non-component scripts should reside? We could certainly point developers towards `frontend/lib/model/` for `Axiom 2` scenarios.

 We also currently we have a number of `lib` folders: `frontend/lib/`, `frontend/web/lib/` and `frontend/web/components/lib` for example, it could be helpful to explain what belongs in each of these directories, although we may want to revisit whether this is the best directory structure before documenting this.

**Comment by SiAdcock** [comment]
@GHaberis Second thoughts: I'd rather not document any of the lib directories for now. I don't think it's a great architecture to have buckets of shareable stuff at every level, and I'd rather not enshrine it in the docs. I think the location of non-component scripts is something that will gradually emerge as we write more of them.

**Comment by GHaberis** [comment]
> the location of non-component scripts is something that will gradually emerge as we write more of them.

I agree this is an area I'm sure we'll revisit in the near future so documenting the current non-ideal situation is probably pointless.

**Pull request merged**